scalameta/dialects/shared/src/main/scala/scala/meta/Dialect.scala

```diff
@@ -74,7 +74,12 @@ final class Dialect private (
     // Extension methods introduced in dotty
     val allowExtensionMethods: Boolean,
     // Open modifier for classes introduced in dotty
-    val allowOpenClass: Boolean
+    val allowOpenClass: Boolean,
+    // Whitebox macros introduced in dotty (splices/quotes)
+    val allowWhiteboxMacro: Boolean,
+    // Top level statements introduced in dotty.
+    // differs from ToplevelTerms because here you can define packages
+    val allowToplevelStatements: Boolean,
```

Comment:
Okay after reading my comment I can see it could be misleading.
TopLevelTerms - enable top level statements, disable packages
TopLevelStatements - enable top level statements, (packages stay enabled by default)
TopLevelTerms and TopLevelStatements should not be used together.
Not sure how to express it better